front/src/modules/simulationResult/components/SpaceTimeChartWrapper/helpers/getPathStyle.ts

```diff
@@ -30,8 +36,19 @@ const getPathStyle = (
   const timetableItemId = isOccurrenceId(train.id)
     ? extractPacedTrainIdFromOccurrenceId(train.id)
     : train.id;
+
   const item = timetableItemsWithDetails?.find((t) => t.id === timetableItemId);
-  const category = item?.category;
+
+  const occurrenceIndex =
+    isOccurrenceId(train.id) && isIndexedOccurrenceId(train.id)
+      ? extractOccurrenceIndexFromOccurrenceId(train.id)
+      : null;
+
+  const exception =
+    item && isPacedTrainWithDetails(item)
+      ? item.exceptions.find((ex) => ex.occurrence_index === occurrenceIndex)
+      : null;
```

Comment:
To handle added exceptions, you can just do this : 

```suggestion
  const exception =
    item && isPacedTrainWithDetails(item) && isOccurrenceId(train.id)
      ? findExceptionWithOccurrenceId(item?.exceptions, train.id)
      : null;
```

Reply:
You're right, it's done